### ISSUE TITLE
Desktop tab nonexistent for remote device running Ubuntu 22.04.1

### ISSUE THREAD

**Issue by markelhosting** [Ylianst/MeshCentral#4421]
Hi all,
I've been having an issue getting the "Desktop" tab to appear for agents that are installed on Ubuntu 22.04.1. I've tried this on a few different Ubuntu installations and get this issue every time. I've ran the `info` command in the console and got "X11 support: false". I've tried installing xorg and other x-related packages, rebooting, etc. and still have this issue. I've tried reinstalling MeshCentral a couple of times, clearing the core and re-uploading the core, hard/soft disconnect, and yet the issue still persists. 

![image](https://user-images.githubusercontent.com/111667809/185771140-67198903-906e-4432-bbbe-1f54b1112abf.png)

![image](https://user-images.githubusercontent.com/111667809/185771197-ad8a49ca-f9ea-4b7f-9cd8-a5b779676eba.png)

![image](https://user-images.githubusercontent.com/111667809/185771288-f5a13b36-6d07-40a8-8fb0-8200ec0fee9b.png)

**Server Software**
 - OS: Ubuntu 22.04.1
 - Virtualization: Oracle Virtualbox
 - Network: LAN/WAN
 - Version: 1.0.70 (tested on previous as well)
 - Node: 12.22.9

**Client Device**
 - Device: Desktop, Mac
 - OS: Windows, masOS Monterey
 - Network: Local to Meshcentral and Remote over WAN
 - Browser: Brave

**Remote Device (please complete the following information):**
 - Device: Ubuntu XCP-ng VMs and Virtualbox VMs, Native Ubuntu installation
 - OS: Ubuntu 22.04.1, earlier version as well
 - Network: Local to Meshcentral and Remote over WAN
 - Current Core Version: Apr 4 2022, 597287619

**config.json file**
```
{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "mesh.mydomain.com",
    "_WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "port": 443,
    "_aliasPort": 443,
    "redirPort": 80,
    "_redirAliasPort": 80,
    "RelayPort": 444
  },
  "domains": {
    "": {
      "title": "mytitle",
      "_title2": "Servername",
      "titlePicture": "title.png",
      "loginPicture": "logo.png",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true,
      "smbios": true
    }
  },
  "letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "service@mydomain.com",
    "names": "mesh.mydomain.com",
    "production": true
  }
}
```

Thanks.

**Comment by jsastriawan** [comment]
Meshcentral Desktop cannot work on gdm/xdm/lightdm login or when the desktop is run on top of Wayland.

Try this step. When you log in to your Ubuntu desktop, choose Ubuntu-Xorg (or similar name) then try Meshcentral desktop connection again. You should be able to access it now.

**Comment by markelhosting** [comment]
@jsastriawan I'm not quite sure what you mean by:
> When you log in to your Ubuntu desktop, choose Ubuntu-Xorg (or similar name) then try Meshcentral desktop connection again.
There's nothing for me to 'choose' when I sign in. You're going to have to dumb it down for me haha.

**Comment by jsastriawan** [comment]
![IMG_20220820_185648.jpg](https://user-images.githubusercontent.com/6583218/185772011-b4b04126-575d-4e56-b625-a960bfa990bc.jpg)

I hope this help.

**Comment by markelhosting** [comment]
![image](https://user-images.githubusercontent.com/111667809/185772046-f59990d2-c3a0-4cc7-96bf-9ce5fc311aad.png)

You da man! It works!

Is there any way (using automatic login) to use "Ubuntu on Xorg"?

**Comment by jsastriawan** [comment]
See https://help.ubuntu.com/stable/ubuntu-help/user-autologin.html.en

**Comment by markelhosting** [comment]
I have it enabled, but when I log off or reboot the machine, will it automatically use "Ubuntu on Xorg"?

**Comment by markelhosting** [comment]
Ahh, got it. Thanks for the edit. I appreciate your help!

**Comment by si458** [comment]
Glad you got it fixes
You can also hard code a ubuntu fix into a file if you need too
https://github.com/Ylianst/MeshCentral/issues/4142#issuecomment-1165845314
@silversword411 could you possibly add this issue to the FAQ about Ubuntu 22.04?
With both the gui fix and the cli fix?
